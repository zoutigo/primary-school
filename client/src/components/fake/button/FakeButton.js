import React from 'react'
import './button.css'

function FakeButton({ label, onClick }) {
  return (
    <div data-testid="button" className="button-style" onClick={onClick}>
      {label} Hello
    </div>
  )
}

export default FakeButton
