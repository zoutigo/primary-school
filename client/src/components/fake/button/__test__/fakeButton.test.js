import React from 'react'
import ReactDom from 'react-dom'
import FakeButton from '../FakeButton'
import { render, cleanup, fireEvent, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

describe('FAKE BUTTON', () => {
  afterEach(cleanup)
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDom.render(<FakeButton></FakeButton>, div)
  })
  it('renders button correctly', () => {
    const { getByTestId } = render(<FakeButton label="Click Me"></FakeButton>)
    expect(getByTestId('button')).toHaveTextContent('Click Me')
  })
  it('renders button correctly', () => {
    const { getByTestId } = render(<FakeButton label="Hello"></FakeButton>)
    expect(getByTestId('button')).toHaveTextContent('Hello')
  })
  it('matches snapshot 1', () => {
    const tree = renderer
      .create(<FakeButton label="Save"></FakeButton>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('matches snapshot 2', () => {
    const tree = renderer
      .create(<FakeButton label="Click Me Now"></FakeButton>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onClick prop when clicked', () => {
    const handleClick = jest.fn()
    render(<FakeButton onClick={handleClick} label="Hello"></FakeButton>)
    fireEvent.click(screen.getByText(/Hello/i))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})
