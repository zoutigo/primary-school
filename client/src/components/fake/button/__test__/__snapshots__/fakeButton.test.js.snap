// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAKE BUTTON matches snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Save
   Hello
</div>
`;

exports[`FAKE BUTTON matches snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Click Me Now
   Hello
</div>
`;

exports[`matches snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Save
   Hello
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Click Me Now
   Hello
</div>
`;
